.pre-commit-config.yaml

```diff
@@ -0,0 +1,47 @@
+---
+repos:
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: 'v0.0.275'
+    hooks:
+      - id: ruff
+        args: [--fix, --exit-non-zero-on-fix]
+  # - repo: https://github.com/psf/black
+  #   rev: 23.3.0
+  #   hooks:
+  #     - id: black
+  #       language_version: python3
```

Comment:
This is a common autoformatter, but might make the PR a bit unwieldly.

Reply:
I've replaced this formatter now to also use `ruff`, through `ruff-format` which has ported `black`'s formatting rules